### ISSUE TITLE
issue with  buildSphere in android.

### ISSUE THREAD

**Issue by KarthikeyanPegasus** [Pana-g/flutter_earth_globe#3]
```
timer = Timer.periodic(const Duration(milliseconds: 10), (Timer p) {
      earthController.setZoom(earthController.zoom + 0.1);

      if (earthController.zoom >= 3) {
        p.cancel();
      }
    });
 ```
 when i tried this, i am getting 
 
![Screenshot_1714384072](https://github.com/Pana-g/flutter_earth_globe/assets/62192093/df6306be-93b8-4a10-a250-6e9d65b717eb)

```
Exception has occurred.
IndexError (RangeError (index): Index out of range: index should be less than 306936: 307227)
```

From This Function:
```
Future<SphereImage?> buildSphere(double maxWidth, double maxHeight) async {
...
...
  spherePixels[(sphereY + x - minX).toInt()] = color;
}
```

**Comment by Pana-g** [comment]
@KarthikeyanPegasus Hmm that's a strange behavior indeed. I'll have a look at it asap.

**Comment by Pana-g** [comment]
@KarthikeyanPegasus Can you provide the full example you are using?

**Comment by Pana-g** [comment]
@KarthikeyanPegasus This issue has been resolved in the latest version 1.0.4.
